app/process/handler/consensus-message/consensus-message.go

```diff
@@ -89,21 +111,95 @@ func (cmh ConsensusMessageHandler) handlePayload(payload []byte) error {
+		TransactionId:        m.TransactionId,
+		EthAddress:           m.EthAddress,
+		Amount:               m.Amount,
+		Fee:                  m.Fee,
+		Signature:            m.Signature,
+		Hash:                 hexHash,
+		SignerAddress:        address.String(),
+		TransactionTimestamp: m.TransactionTimestamp,
 	})
 	if err != nil {
 		return errors.New(fmt.Sprintf("Could not add Transaction Message with Transaction Id and Signature - [%s]-[%s]", m.TransactionId, m.Signature))
 	}
 
-	log.Printf("Successfully verified and persisted TX with ID [%s]\n", m.TransactionId)
+	log.Infof("Successfully verified and persisted TX with ID [%s]", m.TransactionId)
+
+	txSignatures, err := cmh.repository.GetByTransactionId(m.TransactionId, hexHash)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Could not retrieve Transaction Signatures for Transaction [%s]", m.TransactionId))
+	}
+
+	requiredSigCount := len(cmh.validAddresses)/2 + len(cmh.validAddresses)%2
+	log.Infof("Required signatures: [%v]", requiredSigCount)
+
+	txSignatures, err = cmh.repository.GetByTransactionId(m.TransactionId, hexHash)
+	if err != nil {
+		return errors.New(fmt.Sprintf("Could not retrieve Transaction Signatures for Transaction [%s]", m.TransactionId))
+	}
+
+	if len(txSignatures) < requiredSigCount {
+		log.Warnf(fmt.Sprintf("Insignificant amount of Transaction Signatures for Transaction [%s] - [%d] signatures", m.TransactionId, len(txSignatures)))
+		return nil
+	}
+
+	pos := cmh.findMyPosition(txSignatures)
+	if pos == -1 {
+		return errors.New(fmt.Sprintf("Operator is not amongst the potential leaders - [%v]", pos))
+	}
+
+	log.Infof("My position: [%v]", pos)
+
+	now := time.Now()
+	deadline := now.Add(time.Minute * time.Duration(cmh.deadline*pos))
+
+	log.Infof("NOW: [%v]", now)
+	log.Infof("DEADLINE: [%v]", deadline)
+
+	for now.Before(deadline) {
+		if cmh.transactionSent() {
+			return nil
+		}
+		now = time.Now()
+		time.Sleep(10 * time.Second)
+	}
+	log.Infof("I should submit the transaction now!")
+
+	tasm := &validatorproto.TopicAggregatedSignaturesMessage{
+		Hash:      hexHash,
+		EthTxHash: "0x12345",
+	}
+
+	_, err = proto.Marshal(tasm) // topicMessageBytes
+	if err != nil {
+		log.Error("Could not marshal Topic Aggregated Signatures Message [%s] - [%s]", tasm, err)
+	}
+
+	log.Infof("I should send the message! - [%v]", tasm)
+	//_, err = cmh.hederaNodeClient.SubmitTopicConsensusMessage(cmh.topicID, topicMessageBytes)
+	//if err != nil {
+	//	log.Error("Could not Submit Topic Consensus Message [%s] - [%s]", topicMessageBytes, err)
+	//}
+
 	return nil
 }
 
+func (cmh ConsensusMessageHandler) transactionSent() bool {
+	// TODO: Implement Transaction status check
+	return false
+}
+
+func (cmh ConsensusMessageHandler) findMyPosition(messages []message.TransactionMessage) int {
+	sort.Sort(message.ByTimestamp(messages))
+	for i := 0; i < len(messages); i++ {
+		if messages[i].SignerAddress == cmh.operatorAddress {
+			return i
+		}
+	}
+	return -1
```

Comment:
if the operator is not amongst signers, I think that it would be more appropriate if we return an error and check for that error instead of `-1`